source/python/run_scraper.py

```diff
@@ -1,12 +1,26 @@
 import logging
 import sys
+from datetime import timedelta
 
-from sqlalchemy import create_engine
+from sqlalchemy import create_engine, func
 
 from idetect.model import db_url, Base, Session, Status
 from idetect.scraper import scrape
 from idetect.worker import Worker
 
+
+# Filter function for identifying analyses to scrape
+def scraping_filter(query):
+    # Choose either New analyses OR
+    # Analyses where Scraping Failed &
+    # less than 3 scraping attempts &
+    # last scraping attempt greater than 12 hours ago
+    return query.filter((Analysis.status == Status.NEW) |
+                        ((Analysis.status == Status.SCRAPING_FAILED) &
+                         (Analysis.retrieval_attempts < 3) &
```

Comment:
`3` and `timedelta(hours=12)` seem like they should be in a config somewhere.

Reply:
Thanks!
Do you think having a config.py file somewhere and importing that is the right way to go, or would you recommend a different approach?

Reply:
I'd probably start by making these file-local constants, but yeah, eventually a general config.py or config.json makes sense to me. There's no _pressing_ need to split this out, it's just the kind of thing that seems like someone will want to tune at some point.